snapcraft/internal/errors.py

```diff
@@ -133,6 +133,24 @@ def __init__(self, *, base, linker_path):
         super().__init__(base=base, linker_path=linker_path)
 
 
+class IncompatibleBaseError(SnapcraftError):
+    fmt = (
+        "The linker version {linker_version!r} used by the base {base!r} is "
+        "incompatible with all or some of the files in this snap:\n"
+        "{file_list}"
```

Comment:
Does not explain how they might resolve this issue.

Reply:
you missed the conversation; with the build environment stuff coming soon, this should not be raised unless you are in the "I am on my own mode".